ggml/src/ggml-sycl/builtins.hpp

```diff
@@ -0,0 +1,212 @@
+
+using uint16_t2  = vector_t<uint16_t, 2>;
+using uint16_t4  = vector_t<uint16_t, 4>;
+using uint16_t8  = vector_t<uint16_t, 8>;
+using uint16_t16 = vector_t<uint16_t, 16>;
+
+using uint32_t2  = vector_t<uint32_t, 2>;
+using uint32_t8  = vector_t<uint32_t, 8>;
+using uint32_t16 = vector_t<uint32_t, 16>;
+using uint32_t32 = vector_t<uint32_t, 32>;
+
+using int32_t2 = vector_t<int32_t, 2>;
+}  // namespace sycl::vector_types
+
+using coord_t = sycl::vector_types::int32_t2;
+
+namespace sycl::detail {
+
+// Avoid compilation warnings in the host side
+// These are expected to be unused
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-parameter"
+#pragma clang diagnostic ignored "-Wreturn-type"
+
+SYCL_DEVICE_BUILTIN(uint16_t __builtin_IB_subgroup_block_read_flat_u16_m1k16v1(intptr_t baseoffset, int width_minus_one,
+                                                                               int height_minus_one,
+                                                                               int pitch_minus_one, coord_t coord));
+SYCL_DEVICE_BUILTIN(sycl::vector_types::ushort2 __builtin_IB_subgroup_block_read_flat_u16_m2k16v1(
+    intptr_t baseoffset, int width_minus_one, int height_minus_one, int pitch_minus_one, coord_t coord));
+SYCL_DEVICE_BUILTIN(sycl::vector_types::ushort4 __builtin_IB_subgroup_block_read_flat_u16_m4k16v1(
+    intptr_t baseoffset, int width_minus_one, int height_minus_one, int pitch_minus_one, coord_t coord));
+SYCL_DEVICE_BUILTIN(sycl::vector_types::ushort8 __builtin_IB_subgroup_block_read_flat_u16_m8k16v1(
+    intptr_t baseoffset, int width_minus_one, int height_minus_one, int pitch_minus_one, coord_t coord));
+SYCL_DEVICE_BUILTIN(sycl::vector_types::ushort16 __builtin_IB_subgroup_block_read_flat_u16_m16k16v1(
+    intptr_t baseoffset, int width_minus_one, int height_minus_one, int pitch_minus_one, coord_t coord));
+SYCL_DEVICE_BUILTIN(uint32_t __builtin_IB_subgroup_block_read_flat_u32_m1k16v1(
+    intptr_t baseoffset, int width_minus_one, int height_minus_one, int pitch_minus_one, coord_t coord));
+
+SYCL_DEVICE_BUILTIN(int __builtin_IB_dp4a_ss(int c, int a, int b) __attribute__((const)));
+
+#pragma clang diagnostic pop
+
+template <int ElementSize, int BlockWidth, int BlockHeight, int BlockCount> struct XeSubgroup2DBlockLoad {
+    static_assert(false, "Unsupported 2D Block Load Configuration.");
+};
+
+template <> struct XeSubgroup2DBlockLoad<2, 16, 1, 1> {
+    template <typename T>
+    __dpct_inline__ void operator()(const void * srcBasePointer, int memoryWidth, int memoryHeight, int memoryPitch,
+                                    coord_t coordinate, T * dstPointer) {
+#if defined(__SYCL_DEVICE_ONLY__) && !defined(__NVPTX__)
+        *reinterpret_cast<uint16_t *>(dstPointer) = __builtin_IB_subgroup_block_read_flat_u16_m1k16v1(
+            (intptr_t) (srcBasePointer), memoryWidth - 1, memoryHeight - 1, memoryPitch - 1, coordinate);
+#else
+       (void)srcBasePointer;
```

Comment:
Nit: Is it possible to use GGML_UNUSED macro to be more in line with the project?